src/Discord.Net.Core/Entities/SlashCommands/IApplicationCommandOption.cs

```diff
@@ -0,0 +1,40 @@
+using System.Collections.Generic;
+
+namespace Discord
+{
+    /// <summary>
+    /// Represents an Application Command Option, this may be a Sub-Command, a Sub-Command group or a command parameter
+    /// </summary>
+    public interface IApplicationCommandOption
```

Comment:
Deleting the one on my branch but I think instead of the Rest and Socket variants of the Options and Choices, we need to implement shared types in the Core library since these object aren't actually Rest/Socket entities and no operations can be performed on them anyways